beluga_amcl/src/ndt_amcl_node.cpp

```diff
@@ -178,81 +132,52 @@ auto NdtAmclNode::get_initial_estimate() const -> std::optional<std::pair<Sophus
   return std::make_pair(pose, covariance);
 }
 
-auto NdtAmclNode::get_motion_model() const -> MotionModelVariant {
+auto NdtAmclNode::get_motion_model() const -> beluga::DifferentialDriveModel2d {
   const auto name = get_parameter("robot_model_type").as_string();
-  if (name == kDifferentialModelName) {
+  if (name == kDifferentialModelName || name == kNav2DifferentialModelName) {
```

Comment:
@nahueespinosa meta: why adding `kNav2DifferentialModelName` here? The AMCL node needs it to ensure parity with Nav2 AMCL, but the NDT node does not have such constraint.

Reply:
We use `nav2_amcl::DifferentialMotionModel` in the `default_ndt.ros2.yaml` file. Let me change that.